src/Compilers/CSharp/Portable/Compiler/EqualityMethodBodySynthesizer.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Emit
+{
+    internal static class EqualityMethodBodySynthesizer
+    {
+        public static BoundExpression GenerateEqualsComparisons<TList>(
+            NamedTypeSymbol containingType,
+            BoundExpression receiverAccess,
+            TList componentProperties,
```

Comment:
>TList [](start = 12, length = 5)

Why is this generic instead of `IReadOnlyList<FieldSymbol>`, or somthing concrete such as `ImmutableArray<FieldSymbol>`?

Reply:
OK, I see why. The concrete type is `ArrayBuilder` but you don't want to suggest to the caller that this API modifies its input.

---
In reply to: [377359262](https://github.com/dotnet/roslyn/pull/41401#discussion_r377359262) [](ancestors = 377359262)